catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/forms/builder/AttributeValueNormalizer.java

```diff
@@ -0,0 +1,109 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.ui.forms.builder;
+
+import ddf.catalog.data.AttributeDescriptor;
+import ddf.catalog.data.AttributeRegistry;
+import ddf.catalog.data.AttributeType;
+import java.time.DateTimeException;
+import java.time.Instant;
+import java.time.format.DateTimeParseException;
+import java.util.Objects;
+import java.util.Optional;
+import javax.annotation.Nullable;
+import org.codice.ddf.catalog.ui.forms.filter.FilterProcessingException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+class AttributeValueNormalizer {
+  private static final Logger LOGGER = LoggerFactory.getLogger(AttributeValueNormalizer.class);
+
+  private final AttributeRegistry registry;
+
+  AttributeValueNormalizer(AttributeRegistry registry) {
+    this.registry = registry;
+  }
+
+  String normalizeForJson(String property, String value) {
+    if (isNotNormalizableDateValue(property, value)) {
+      return value;
+    }
+    Instant instant = instantFromEpoch(value);
+    if (instant != null) {
+      return Objects.toString(instant);
+    }
+    return value;
+  }
+
+  String normalizeForXml(String property, String value) {
```

Comment:
Just an FYI, we have a fairly robust date parser included in EndpointUtil because of variability we've encountered in the past 😄 
https://github.com/codice/ddf/blob/master/catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/util/EndpointUtil.java#L673-L724

Reply:
Thanks for the heads up. I'll look into this.

Reply:
So I checked out the parsing function in endpoint util and it turns out I need different semantics than what it offers. So I just added support for offsets to my code without adopting the util to get the functionality I needed.